src/vm/ceemain.cpp

```diff
@@ -2176,7 +2167,7 @@ void STDMETHODCALLTYPE EEShutDown(BOOL fIsDllUnloading)
     }
 
 #ifdef FEATURE_COMINTEROP
-    if (!fIsDllUnloading && IsThreadInSTA())
+    if (!fIsDllUnloading && IsThreadInSTA() && CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_FinalizeOnShutdown))
```

Comment:
The `EXTERNAL_FinalizeOnShutdown` check should be done first, before `IsThreadInSTA`.